ngraph/core/src/op/interpolate.cpp

```diff
@@ -111,143 +116,485 @@ namespace ngraph
+            scales.insert(scales.end(), scales_ptr, scales_ptr + num_of_axes);
+        }
+        else
+        {
+            auto target_shape = get_target_shape_vector(args, num_of_axes);
+            for (size_t i = 0; i < num_of_axes; ++i)
+            {
+                size_t axis = axes[i];
+                float scale =
+                    static_cast<float>(target_shape[i]) / static_cast<float>(input_shape[axis]);
+                scales.push_back(scale);
+            }
+        }
+        return scales;
+    }
+
+    template <typename T>
+    std::vector<T> correct_pad(const std::vector<T>& p, size_t rank)
+    {
+        size_t pad_len = p.size();
+        if (pad_len == rank)
+        {
+            return p;
+        }
+
+        std::vector<T> result;
+
+        if (pad_len > rank)
+        {
+            result.insert(result.end(), p.begin(), p.begin() + rank);
+        }
+        else
+        {
+            result = p;
+            result.insert(result.end(), rank - pad_len, T{});
+        }
+
+        return result;
+    }
+}
+
+void op::v4::Interpolate::correct_pads()
+{
+    PartialShape input_shape = PartialShape(get_input_partial_shape(0));
+    if (input_shape.rank().is_dynamic())
+    {
+        return;
+    }
+    const auto input_rank = input_shape.rank().get_length();
+
+    m_attrs.pads_begin = correct_pad(m_attrs.pads_begin, input_rank);
+    m_attrs.pads_end = correct_pad(m_attrs.pads_end, input_rank);
+}
+
+static void pad_input_data(const uint8_t* data_ptr,
+                           uint8_t* padded_data_ptr,
+                           size_t type_size,
+                           const Shape& input_shape,
+                           const Shape& padded_input_shape,
+                           const std::vector<size_t>& pads_begin)
+{
+    CoordinateTransform input_transform(input_shape);
+    CoordinateTransform padded_transform(padded_input_shape);
+
+    for (const Coordinate& input_coord : input_transform)
+    {
+        auto padded_coord = input_coord;
+        size_t i = 0;
+        for (size_t pad : pads_begin)
+        {
+            padded_coord[i] += pad;
+            ++i;
+        }
+        uint8_t* dst_ptr = padded_data_ptr + type_size * padded_transform.index(padded_coord);
+        const uint8_t* src_ptr = data_ptr + type_size * input_transform.index(input_coord);
+        memcpy(dst_ptr, src_ptr, type_size);
+    }
+}
+
+bool op::v4::Interpolate::evaluate(const HostTensorVector& outputs,
+                                   const HostTensorVector& inputs) const
+{
+    element::Type input_et = get_input_element_type(0);
+    size_t type_size = input_et.size();
+
+    Shape input_shape{inputs[data_port]->get_shape()};
+    Shape padded_input_shape = get_padded_input_shape(input_shape).to_shape();
+
+    auto axes = get_axes_vector(inputs);
+    size_t num_of_axes = axes.size();
+
+    auto scales = get_scales_vector(inputs, input_shape, m_attrs, axes);
```

Comment:
We actually resize padded_input_shape tensor to target_shape tensor as I refer the spec. Scales calculation should be done with these two values. There are no pad process within get_scales_vector(), so here should be  `"auto scales = get_scales_vector(inputs, padded_input_shape , m_attrs, axes);"`

Reply:
Fixed.